client/state/initial-state.js

```diff
@@ -104,8 +105,13 @@ function verifyStateTimestamp( state ) {
 	return state._timestamp && state._timestamp + MAX_AGE > Date.now();
 }
 
-export async function getStateFromLocalStorage( reducer, subkey ) {
-	const reduxStateKey = getReduxStateKey() + ( subkey ? ':' + subkey : '' );
+export async function getStateFromLocalStorage( reducer, subkey, forceLoggedOutUser = false ) {
+	let reduxStateKey;
+	if ( forceLoggedOutUser ) {
+		reduxStateKey = 'redux-state-logged-out' + ( subkey ? ':' + subkey : '' );
+	} else {
+		reduxStateKey = getReduxStateKey() + ( subkey ? ':' + subkey : '' );
```

Comment:
It makes sense to add `forceLoggedOutUser` parameter to `getReduxStateKey` and then call it as:
```js
const reduxStateKey = getReduxStateKey( forceLoggedOutUser ) + ( subkey ? ':' + subkey : '' );
```

Reply:
Good idea, I've done this in my separate PR for the initial state changes: https://github.com/Automattic/wp-calypso/pull/34982